docs/contributing/coding-guidelines/ros-nodes/console-logging.md

```diff
@@ -1,5 +1,183 @@
+
+```xml
+<launch>
+  <!-- This outputs to rosout and disk. -->
+  <node pkg="demo_nodes_cpp" exec="talker" ros_args="--disable-stdout-logs" />
+</launch>
+```
+
+```xml
+<launch>
+  <!-- This outputs to stdout. -->
+  <node pkg="demo_nodes_cpp" exec="talker" ros_args="--disable-rosout-logs --disable-external-lib-logs" />
+</launch>
+```
+
+### Use throttled logging when the log is unnecessarily shown repeatedly (required, non-automated)
+
+#### Rationale
+
+If tons of logs are shown on the console, people miss important message.
+
+#### Example
+
+While waiting for some messages, throttled logs are usually enough.
+In such cases, wait about 5 seconds as a reference value.
+
+```cpp
+// Compliant
+void FooNode::on_timer() {
+  if (!current_pose_) {
+    RCLCPP_ERROR_THROTTLE(get_logger(), *get_clock(), 5000, "Waiting for current_pose_.");
+    return;
+  }
+}
+
+// Non-compliant
+void FooNode::on_timer() {
+  if (!current_pose_) {
+    RCLCPP_ERROR(get_logger(), "Waiting for current_pose_.");
+    return;
+  }
+}
+```
+
+#### Exception
+
+The following cases are acceptable even if it's not throttled.
+
+- The message is really worth displaying every time.
+- The message level is DEBUG.
+
+### Do not depend on rclcpp::Node in core library classes but depend only on rclcpp/logging.hpp (advisory, non-automated)
+
+#### Rationale
+
+Core library classes, which contain reusable algorithms, may also be used for non-ROS platforms.
+When porting libraries to other platforms, fewer dependencies are preferred.
+
+#### Example
+
+```cpp
+// Compliant
+#include <rclcpp/logging.hpp>
+
+class FooCore {
+public:
+  explicit FooCore(const rclcpp::Logger & logger) : logger_(logger) {}
+
+  void process() {
+    RCLCPP_INFO(logger_, "message");
+  }
+
+private:
+  rclcpp::Logger logger_;
+};
+
+// Compliant
+// Note that logs aren't published to `/rosout` if the logger name is different from the node name.
+#include <rclcpp/logging.hpp>
+
+class FooCore {
+  void process() {
+    RCLCPP_INFO(rclcpp::get_logger("foo_core_logger"), "message");
+  }
+};
+
+
+// Non-compliant
+#include <rclcpp/node.hpp>
+
+class FooCore {
+public:
+  explicit FooCore(const rclcpp::NodeOptions & node_options) : node_("foo_core_node", node_options) {}
+
+  void process() {
+    RCLCPP_INFO(node_.get_logger(), "message");
+  }
+
+private:
+  rclcpp::Node node_;
+};
+```
+
+## Tips
+
+### Use rqt_console to filter logs
+
+To filter logs, using `rqt_console` is useful:
+
+```bash
+ros2 run rqt_console rqt_console
+```
+
+For more details, refer to [ROS 2 Documentation](https://docs.ros.org/en/rolling/Tutorials/Beginner-CLI-Tools/Using-Rqt-Console/Using-Rqt-Console.html).
+
+### Useful marco expressions
+
+To debug program, sometimes you need see which functions and lines of code are executed.
+In that case, use `__LINE__` and `__FUNCTION__` macro:
```

Comment:
FYI, I also use `__FILE__` with `__LINE__`, but I will leave it up to you if you want to include it in this chapter.

Reply:
Thanks. Updated in 95cc408f37a995c6b62b0d8b9b3d6ea2d8ca0152.